Purchases/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -153,14 +153,22 @@ fileprivate extension TrialOrIntroPriceEligibilityChecker {
 
 }
 
-private extension TrialOrIntroPriceEligibilityChecker {
+internal extension TrialOrIntroPriceEligibilityChecker {
 
     @available(iOS 11.2, macOS 10.13.2, macCatalyst 13.0, tvOS 11.2, watchOS 6.2, *)
-    func productsWithIntroOffers(productIdentifiers: [String], completion: @escaping ReceiveIntroEligibilityBlock) {
+    func mapProductsWithIntroEligibility(productIdentifiers: [String],
+                                         completion: @escaping ReceiveIntroEligibilityBlock) {
         self.productsManager.products(withIdentifiers: Set(productIdentifiers)) { products in
             let eligibility: [(String, IntroEligibility)] = Array(products.value ?? [])
-                .filter { $0.introductoryDiscount != nil }
-                .map { ($0.productIdentifier, IntroEligibility(eligibilityStatus: .eligible)) }
+                .map({ storeProduct in
+                    let status: IntroEligibilityStatus
+                    if storeProduct.introductoryDiscount == nil {
+                        status = .noIntroOfferExists
+                    } else {
+                        status = .eligible
+                    }
```

Comment:
This seems to be used for StoreKit 1 with iOS 11.2 but below iOS 12... unless there is an error in iOS 12 and up and then this is used as a secondary check